src/java.base/share/native/libjli/parse_manifest.c

```diff
@@ -418,7 +497,32 @@ find_file(int fd, zentry *entry, const char *file_name)
          */
         if ((size_t)CENNAM(p) == JLI_StrLen(file_name) &&
           memcmp((p + CENHDR), file_name, JLI_StrLen(file_name)) == 0) {
-            if (JLI_Lseek(fd, base_offset + CENOFF(p), SEEK_SET) < (jlong)0) {
+            jlong cenlen = CENLEN(p);
+            jlong censiz = CENSIZ(p);
+            jlong cenoff = CENOFF(p);
+            jlong cenext = CENEXT(p);
+            if ((cenlen == ZIP64_MAGICVAL
+                  || censiz == ZIP64_MAGICVAL
+                  || cenoff == ZIP64_MAGICVAL)
+                && cenext > 0) {
+              // The extra fields start after the fixed-size central directory
+              // header and the variable-length file name.
+              Byte *base = p + CENHDR + CENNAM(p);
+              jlong offset = 0;
+              // Scan the extra fields for zip64 extra info, see APPNOTE.TXT 4.5
+              while (offset < cenext) {
+                short headerId = ZIPEXT_HDR(base + offset);
+                short headerSize = ZIPEXT_SIZ(base + offset);
+                if (headerId == ZIP64_EXTID) {
+                  if (!read_zip64_ext(base + offset, &cenlen, &censiz, &cenoff, CENDSK(p))) {
+                    free(buffer);
+                    return (-1);
+                  }
```

Comment:
I think we are missing a `break;` here, after this inner `if` block reads the zip64 extra fields and returns `JNI_TRUE`.

Reply:
Thanks for the catch, fixed. It is unnecessary to loop over any other extensions that are present after the zip64 one is found.